test/integration/weighted_cluster_integration_test.cc

```diff
@@ -0,0 +1,190 @@
+        auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+        cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+        cluster->mutable_load_assignment()
+            ->mutable_endpoints(0)
+            ->mutable_lb_endpoints(0)
+            ->mutable_endpoint()
+            ->mutable_address()
+            ->mutable_socket_address()
+            ->set_address(fake_upstreams_[i]->localAddress()->ip()->addressAsString());
+        cluster->set_name(absl::StrFormat(Envoy::RepickClusterFilter::ClusterName, i));
+        ConfigHelper::setHttp2(*cluster);
+      }
+
+      auto* cluster_with_name = bootstrap.mutable_static_resources()->mutable_clusters(0);
+      cluster_with_name->set_name(absl::StrFormat(Envoy::RepickClusterFilter::ClusterName, 0));
+      ConfigHelper::setHttp2(*cluster_with_name);
+
+      // Add the custom filter.
+      config_helper_.addFilter("name: repick-cluster-filter");
+    });
+
+    // Modify route with weighted cluster configuration.
+    config_helper_.addConfigModifier(
+        [&weights](
+            envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+                hcm) {
+          for (int i = 0; i < TotalUpstreamClusterWithNameCount; ++i) {
+            auto* cluster_with_name = hcm.mutable_route_config()
+                                          ->mutable_virtual_hosts(0)
+                                          ->mutable_routes(0)
+                                          ->mutable_route()
+                                          ->mutable_weighted_clusters()
+                                          ->add_clusters();
+            cluster_with_name->set_name(
+                absl::StrFormat(Envoy::RepickClusterFilter::ClusterName, i));
+            cluster_with_name->mutable_weight()->set_value(weights[i]);
+          }
+
+          for (int i = TotalUpstreamClusterWithNameCount;
+               i < Envoy::RepickClusterFilter::TotalUpstreamClusterCount; ++i) {
+            auto* cluster_with_header = hcm.mutable_route_config()
+                                            ->mutable_virtual_hosts(0)
+                                            ->mutable_routes(0)
+                                            ->mutable_route()
+                                            ->mutable_weighted_clusters()
+                                            ->add_clusters();
+            cluster_with_header->set_cluster_header(
+                absl::StrFormat(Envoy::RepickClusterFilter::ClusterHeaderName, i));
+            cluster_with_header->mutable_weight()->set_value(weights[i]);
+          }
+
+          hcm.mutable_route_config()
+              ->mutable_virtual_hosts(0)
+              ->mutable_routes(0)
+              ->mutable_route()
+              ->mutable_weighted_clusters()
+              ->mutable_total_weight()
+              ->set_value(std::accumulate(weights.begin(), weights.end(), 0UL));
+        });
```

Comment:
I did some quick look/search, but i am not sure if fmt::format is a good choice here because:
1. It seems that templated (via fmt::format) yaml string will require more work to spell out virtual_hosts configuration (e.g. name, domains, match etc, which are default in integration test framework.)
2. I searched `addConfigModifier` usage, all the occurrence seems are using callback method.

I used yaml string in unit test (i.e. config_impl_test) where it has simple configuration. But I am not familiar with `templated (via fmt::format) yaml string` and would like to learn your thoughts. Thanks!